ethdb/remote/remotedbserver/ethbackend.go

```diff
@@ -55,3 +59,31 @@ func (s *EthBackendServer) NetVersion(_ context.Context, _ *remote.NetVersionReq
 	}
 	return &remote.NetVersionReply{Id: id}, nil
 }
+
+func (s *EthBackendServer) Subscribe(_ *remote.SubscribeRequest, subscribeServer remote.ETHBACKEND_SubscribeServer) error {
+	log.Debug("establishing event subscription channel with the RPC daemon")
+	wg := sync.WaitGroup{}
+	wg.Add(1)
+
+	s.events.AddHeaderSubscription(func(h *types.Header) error {
+		payload, err := json.Marshal(h)
+		if err != nil {
+			log.Warn("error while marshaling a header", "err", err)
+			return err
+		}
+
+		err = subscribeServer.Send(&remote.SubscribeReply{
+			Type: uint64(EventTypeHeader),
+			Data: payload,
+		})
+
+		if err != nil {
+			wg.Done()
```

Comment:
Why is `wg.Done` not done in the case of success?

Reply:
because io.EOF is an error and if it is a success that means that the channel is not closed and we need to keep reading from it.

Reply:
but good point, I'll add a comment about that